libs/dh/shared/data-access-mocks/src/lib/admin.ts

```diff
@@ -67,150 +69,160 @@ export function adminMocks(apiBase: string) {
         (userRole) => userRole.id === userRoleId
       );
-
-      return res(ctx.json(userRole));
+      await delay(mswConfig.delay);
+      return HttpResponse.json(userRole);
     }
   );
 }
 
 function getUserRoleAuditLogs() {
-  return mockGetUserRoleAuditLogsQuery((req, res, ctx) => {
-    return res(ctx.data(getUserRoleAuditLogsMock));
+  return mockGetUserRoleAuditLogsQuery(async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json({ data: getUserRoleAuditLogsMock });
   });
 }
 
 function putMarketParticipantUserRoleUpdate(apiBase: string) {
-  return rest.put(`${apiBase}/v1/MarketParticipantUserRole/Update`, (req, res, ctx) => {
-    return res(ctx.status(200));
+  return http.put(`${apiBase}/v1/MarketParticipantUserRole/Update`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(null, { status: 200 });
   });
 }
 
 function getMarketParticipantOrganizationGetFilteredActors(apiBase: string) {
-  return rest.get(
-    `${apiBase}/v1/MarketParticipant/Organization/GetFilteredActors`,
-    (req, res, ctx) => {
-      return res(ctx.json(marketParticipantOrganizationGetFilteredActors));
-    }
-  );
+  return http.get(`${apiBase}/v1/MarketParticipant/Organization/GetFilteredActors`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(marketParticipantOrganizationGetFilteredActors);
+  });
 }
 
 function getAdminPermissions() {
-  return mockGetPermissionsQuery((req, res, ctx) => {
-    return res(ctx.delay(300), ctx.data(adminPermissionsMock));
+  return mockGetPermissionsQuery(async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json({ data: adminPermissionsMock });
   });
 }
 
 function getAdminPermissionDetails() {
-  return mockGetPermissionDetailsQuery((req, res, ctx) => {
-    return res(ctx.delay(300), ctx.data(adminPermissionDetailsMock));
+  return mockGetPermissionDetailsQuery(async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json({ data: adminPermissionDetailsMock });
   });
 }
 
 function getAdminPermissionLogs() {
-  return mockGetPermissionAuditLogsQuery((req, res, ctx) => {
-    const permId = req.variables.id;
+  return mockGetPermissionAuditLogsQuery(async ({ variables }) => {
+    const permId = variables.id;
     const permissionAuditLogs = [adminPermissionAuditLogsMock[permId]];
-    return res(ctx.delay(300), ctx.data({ __typename: 'Query', permissionAuditLogs }));
+    await delay(mswConfig.delay);
+    return HttpResponse.json({ data: { __typename: 'Query', permissionAuditLogs } });
   });
 }
 
 function putMarketParticipantPermissionsUpdate(apiBase: string) {
-  return rest.put(`${apiBase}/v1/MarketParticipantPermissions/Update`, (req, res, ctx) => {
-    return res(ctx.status(200));
+  return http.put(`${apiBase}/v1/MarketParticipantPermissions/Update`, async () => {
+    await delay(mswConfig.delay);
+    return HttpResponse.json(null, { status: 200 });
```

Comment:
suggestion: the migration guide for MSW v2 uses `new HttpResponse()` for mocking a status-only requests. https://mswjs.io/docs/migrations/1.x-to-2.x#ctxstatus

```suggestion
    return new HttpResponse(null, { status: 200 });
```